pubsub.go

```diff
@@ -72,6 +75,18 @@ type PubSub struct {
 	// a notification channel for when our peers die
 	peerDead chan peer.ID
 
+	// tracks the channels subscribed to new peer notifications
+	newPeerNotif map[chan peer.ID]struct{}
+
+	// setup new peer notifier
+	addPeerNotif chan *reqNewPeerNotifier
+
+	// remove peer notifier
+	rmPeerNotif chan chan peer.ID
+
+	// peers that we have connected to, but have not received a Hello from
+	inflightPeers map[peer.ID]struct{}
```

Comment:
no, we shouldn't be tracking those. The connection notifications will issue hello.

Reply:
we track successful connections, but not really unsuccessful ones. We need the inflight peers to properly handle those.

Reply:
again, this is leaking through the abstraction barrier -- it's irrelevant to the mainloop.